week1-IntroAndSetupGuide.md

```diff
@@ -0,0 +1,49 @@
+
+With your repo name in hands (it will be your working repo for the duration of the project), let's get started! 
+
+ 1. Create a local folder to keep your code. Ex: C:\CodessInTheClassroom.
+ 2. Find your repo at [CodessInTheClassrom GitHub project](https://github.com/codessintheclassroom).
+ 3. Clone your repository: instructions available [here](git-cheatsheet.md#clone).  
+
+At this stage you have your code checked out locally, and by default you would be working off the `master` branch. We should avoid committing and pushing code straight to `master`, as this branch is usually where your stable code should be (we normally use `master` as a source for production code). We'll talk about this in a bit more detail when we discuss Pull Requests. 
+Having that in mind, we will now:
+
+ 1. Create a new branch for the next coding steps.    
+  1.1. Make sure you have a descriptive branch name. Ex: `your-username/workshop1-adding-react-app`.  
+  1.2. Check the [git cheatsheet](git-cheatsheet.md#create-a-branch) for instructions.
+ 2. Install the latest Long Term Support (LTS) version of node js from https://nodejs.org/en/. Currently this is version 10.  
+  2.1. Avoid non LTS versions, e.g. version 11, since they tend to have bugs.  
+
+## Starting the app 
+ This section will give you the steps to generate the base code to start our application. 
+ We'll start by creating a [React](https://reactjs.org/) app: 
+ 
+1. Launch Visual Studio Code and use `File -> Open Folder` and open the folder `front-end`. It will be under the folder we created on the previous sections (Ex: CodessInTheClassroom).  
+2. Open a terminal by clicking on `Terminal -> New Terminal`. 
+3. Create and start a basic typescript based web application. Run the following commands:
+	```
+	 > npx create-react-app web-application --typescript
+	 > cd web-application
+	 > npm start
+	```
+	 See [create-react-app on github](https://facebook.github.io/create-react-app/docs/getting-started)  for more info.
+4. Navigate to http://localhost:3000/ to see the running application.
+
+### Commit and create a PullRequest
+
+So far, we've generated a basic react app, which we will build on, to create the Pet Adoption Service front-end. If you look at your source code now, you'll see a lot of files that were generated by the tools and which are simply boilerplate code: stuff we usually don't change, but yet necessary for the app to run (kind of like the nuts and bolts of our system).
+We will commit and push this code to our master branch now for two reasons: 1. This is a checkpoint for us, meaning we have stable code which starts up a basic react app and 2. From now on, we will start customizing this code to our needs, so we want to have a clear distinction from the code that we generated from the third party tools and the code we'll create as part of the requirements of our system.
```

Comment:
I really like that you've explained not just _what_ we're doing at this point, but _why_ we're doing it. :100: